### ISSUE TITLE
Weird fsync issue

### ISSUE THREAD

**Issue by kaasknak** [leycec/fsnrnue#27]
I know I know. Normally I am not the one opening issues but here we are.

It seems I suddenly get an error when starting the game. It says my kernel isn't patch for fsync and crashes.

Disabling fsync solves it but I am opening an issue nonetheless so that if more people have the same problem I hope we can find a common reason / fix.

OS: Ubuntu 20.04.4
Kernel: 5.13.0-40-generic x86_64

**Comment by leycec** [comment]
That's... **horrible.** Ordinarily, I'd silently walk by while live-streaming your ongoing trainwreck. I'm that kinda jerk. But I can't help myself. Let's be a white knight, guys!

Do you happen to have a traceback or other visible error? It *kinda* seems likely this is a WINE desynchronization issue, going on nothing burgers and gut feels. If so, having to reinstall the entire game would be admittedly awful. Perhaps something more sensible can be done.

Relatedly, Ubuntu 22.04 LTS **Jammy Jellyfish** is officially out! You might try upgrading to that first and seeing if that doesn't solve your world of hurt.

**Comment by kaasknak** [comment]
> Do you happen to have a traceback or other visible error?

Nope. Everything looked clean and normal after I turned fsync off. It is a really weird error because I didn't change anything about the game and it just suddenly gave me that error. I fear indeed WINE shenanigans or that some kind of dependency got updated to something less stable _shudders_.

> Relatedly, Ubuntu 22.04 LTS Jammy Jellyfish is officially out!

It is indeed and while the name Jammy Jellyfish is awesome I am not too happy with many of the other decisions made so I am on the fence on upgrading. I could upgrade and try to make it fit into how I want it to act or I can switch to Linux Mint Mate edition. I am not sure yet but I still have a couple of years of support left to figure it out (probably going to upgrade way before the support ends though).

**Comment by leycec** [comment]
> Nope. Everything looked clean and normal after I turned fsync off.

:face_exhaling: 

> It is indeed and while the name Jammy Jellyfish is awesome I am not too happy with many of the other decisions made so I am on the fence on upgrading.

**Fascinating.** Please tell me more, because I am ignorant.

I'm hip-deep in Gentoo Linux Land, where the price for getting everything you want is only a lifetime of inscrutable low-level configuration and compilation times beyond any pain you have previously known. *But it's all worth it.* Because I get to loudly screech: "Suck it, `systemd`! I'll install you when I jolly want to and not a femtosecond sooner."

My wife, however, wades in the turbid waters of Ubuntu Linux Land. Neither of us read the muddy tea leaves of modern open-source news feeds, so... I didn't even know there was community uproar until this. I mean, I'm hardly surprised. Canonical whiplashes their entire stack with every new LTS. How bad is it this time? I have my popcorn pre-popped and ready, if that helps.

> Linux Mint

`</shudders_spasmodically>`

This was the thing my wife used *before* Ubuntu. She loved the UI. Something spicy that Dutch sailors used to murder each other over like "Cinnamon," I believe it was?

But we both abhorred their upgrade path: **because there was no upgrade path.** Literally, the upgrade path was "format and reinstall your entire machine every year." We stopped doing that fairly quickly, because there is this thing vaguely called Life and we intended to do that rather than bludgeoning Linux Mint into submission.

**Comment by kaasknak** [comment]
Personally I don't like GNOME3 stuff. It tends to be unstable as hell and being able to configure anything in it seems to be impossible without adding a whole plug-in manager which will then interact directly with the program (no API) which means all your plug-ins will probably break every update cycle. While I am not interested enough in configuring every minor detail on my daily driver (because of this thing called life indeed) I do like to be able to customise to some extent. This is why I used Ubuntu Mate (I absolutely love the GNOME2 look and the high degree of customisability while still remaining stable).

Snaps. Unlike many people I don't just hate snaps (or flatpaks) for the sake of hating them. I think both are very nice pieces of software that can solve dependency issues for old software that is no longer being maintained without people having to dig through the internet to find this one obscure version of some dependency that some legacy code is dependent on. HOWEVER. Putting everything in a snap or flatpak is just silly and I really don't get the point of it. If I wanted to fill my hard drive with super bloaty programs I would just run windows. Also snaps have been unstable as hell for me and in the past software in a snap very often just crashed on opening. When using debs for the same program it works flawlessly.
All of this is fine if I could just chose to not use snaps but it seems that Canonical has decided to ship firefox as a snap only. Not only is this problematic because many people in the community are talking about up to a minute of waiting before the browser starts it also tells me that if they are willing to ship the browser as a snap a lot more software will follow.

Ubuntu mate comes with snaps and flatpak and GNOME3 apps (for some reason).
In general it feels a bit like this XKCD comic https://xkcd.com/927/ In order to fix all the different package managers on different distributions we now require you to have three different package managers.

I am still very much on the fence about Linux Mint. It also comes with flatpaks by default (but at least I could remove it) and I have been with Ubuntu continuously since 16.04 so I did grow a bit attached to it.

The upgrade path is not the biggest issue for me. I never trust installers to upgrade things properly so I always just reinstalled.

Gentoo honestly seems like a lot of fun and I would like to play around with it but I am not brave enough to run that on my main machine. I like stability too much for that and I am not educated enough in the arcane language of the compilers.

**Comment by leycec** [comment]
> Gentoo honestly seems like a lot of fun...

![one of us one of us](https://c.tenor.com/SD-5sdtUX8QAAAAC/simpsons-homer.gif)

> I like stability too much for that...

**Hah!** Gentoo is, by a hefty margin, the stablest distro I've ever used – because I run the default `stable` profile. My machine uptime is not to be trifled with. Meanwhile, my wife's Ubuntu laptop rapidly disembowels itself every five minutes over there.

The tradeoff with Gentoo isn't actually stability or security; we've got that covered. That's why most Gentoo users end up running Gentoo as a public-facing (web)server at one point or another. It's profitable to do so, so that's where the ~~spice~~ money flows.

The tradeoff with Gentoo is **your time.** In return, Gentoo gives you ultimate power. Don't like GNOME3? That's great. We support every other window manager. Don't like `systemd`? That's great. We support every other init system. For most of us, Gentoo is some combination of our singular hobby and/or profession – so that tradeoff is fine. We're the automotive enthusiast equivalent of home car modders.

This is why Gentoo is commonly referred to as a [meta-distribution](https://www.gentoo.org/get-started/about). Gentoo effectively builds your own Linux distribution unique to your specific needs. `</mic_drop>`

> Personally I don't like GNOME3 stuff...

**Phenomenal writeup.** Thanks so much for that deep dive into the horrors of modern desktop design! I fully agree, of course. In my case, it's GNOME's lack of configurability that's the real anti-killer app here.

Does it get more bare-bones than the Ubuntu 20.04 system settings panel? It does not. When you have to install a medley of cranky third-party utilities (e.g., Gnome Tweak Tool) just to access what would otherwise be core configuration settings under any other windowing manager, you know you done wrong.

Of course, I use FVWM – an ancient meta-windowing manager <sup>*...there is a theme here*</sup> that requires you to literally code your entire windowing manager from the ground up. With great power comes a great time sink. Shoot me, fam.

> Canonical has decided to ship firefox as a snap only.

...and that's a hard **GTFO** for me. Clearly, they're incrementally migrating from APT to Snap because doing so minimizes their packaging effort. When everything is a Snap, Canonical no longer needs to concern themselves with the maintenance burden that plagues other Linux distributions. Packaging conflicts, interdependencies, and bitrot will then be a thing of the halcyon past. This makes sense from their perspective – while making utterly no sense from the end user perspective, because I have yet to use a Snap that wasn't inferior in every possible way across the multiverse to a standard desktop app with standard user privileges.

**tl;dr:** *Mark Shuttleworth gonna Mark Shuttleworth.*

> In general it feels a bit like this XKCD comic https://xkcd.com/927/ 

I'm a simple man. I see a reference to my favourite XKCD comic, I upvote.

Thanks again for all the earnest contributions at this repo. May the dust eventually settle on your distro choice and your future be bright with Linux-compatible visual novels. Steam Deck: *your time has come.*

**Comment by kaasknak** [comment]
It turns out this is something new in lutris and is possible from linux kernel 5.16 and beyond. Since I am running on an LTS my kernel isn't patched for this.

I'll add another part to the README about this.
